core/src/ibc/channel_04/manager.rs

```diff
@@ -101,4 +103,103 @@ impl<'a> Manager<'a> {
 
         Ok(key)
     }
+
+    pub fn chan_open_try(
+        &mut self,
+        order: ChannelOrder,
+        connection: Identifier,
+        channel_identifier: Identifier,
+        counterparty_channel_identifier: Identifier,
+        version: String,
+        counterparty_version: String,
+        proof_init: Bytes,
+        proof_height: u64,
+    ) -> Result<Identifier, String> {
+        let kv_store = self.ctx.get_kv_store();
+
+        let previous = kv_store.get(&channel_path(&channel_identifier, DEFAULT_PORT));
+
+        if let Some(x) = previous {
+            let channel_end: ChannelEnd = rlp::decode(&x).expect("Illformed connection end stored in the DB");
+            if channel_end.state != ChannelState::INIT
```

Comment:
How about creating an expected channel and comparing the expected and `channel_end`?